homeassistant/components/bluecurrent/__init__.py

```diff
@@ -0,0 +1,244 @@
+                self.add_charge_point(evse_id, model)
+                await self.get_charge_point_data(evse_id)
+            await self.client.get_grid_status(data[0][EVSE_ID])
+
+        object_name: str = message[OBJECT]
+
+        if DATA in message:
+            data: dict | list = message[DATA]
+
+        # gets charge point ids
+        if object_name == CHARGE_POINTS:
+            assert isinstance(data, list)
+            await handle_charge_points(data)
+
+        # gets charge point key / values
+        elif object_name in VALUE_TYPES:
+            assert isinstance(data, dict)
+            evse_id = data.pop(EVSE_ID)
+            self.update_charge_point(evse_id, data)
+
+        # gets grid key / values
+        elif object_name == GRID_STATUS:
+            assert isinstance(data, dict)
+            self.grid = data
+            self.dispatch_signal()
+
+    async def get_charge_point_data(self, evse_id: str) -> None:
+        """Get all the data of a charge point."""
+        await self.client.get_status(evse_id)
+
+    def add_charge_point(self, evse_id: str, model: str) -> None:
+        """Add a charge point to charge_points."""
+        self.charge_points[evse_id] = {MODEL_TYPE: model}
+
+    def update_charge_point(self, evse_id: str, data: dict) -> None:
+        """Update the charge point data."""
+        for key in data:
+            self.charge_points[evse_id][key] = data[key]
+        self.dispatch_signal(evse_id)
+
+    def dispatch_signal(self, evse_id: str | None = None) -> None:
+        """Dispatch a signal."""
+        if evse_id is not None:
+            async_dispatcher_send(self._hass, f"{DOMAIN}_value_update_{evse_id}")
+        else:
+            async_dispatcher_send(self._hass, f"{DOMAIN}_grid_update")
+
+    async def start_loop(self) -> None:
+        """Start the receive loop."""
+        try:
+            await self.client.start_loop(self.on_data)
+        except (BlueCurrentException) as err:
+            LOGGER.warning(
+                "Disconnected from the Blue Current websocket. Retrying to connect in background. %s",
+                err,
+            )
+
+            async_call_later(self._hass, DELAY_1, self.reconnect)
+
+    async def reconnect(self, event_time: datetime | None = None) -> None:
+        """Keep trying to reconnect to the websocket."""
+        try:
+            await self.connect(self._config.data[CONF_API_TOKEN])
+            LOGGER.warning("Reconnected to the Blue Current websocket")
+            self._hass.loop.create_task(self.start_loop())
+            await self.client.get_charge_points()
+        except (WebsocketException, RequestLimitReached) as err:
+            set_entities_unavalible(self._hass, self._config.entry_id)
+            if isinstance(err, RequestLimitReached):
+                delay = self.client.get_next_reset_delta()
+            else:
+                delay = DELAY_2
+            async_call_later(self._hass, delay, self.reconnect)
```

Comment:
Would probably be more readable to split these:
```suggestion
        except RequestLimitReached:
            set_entities_unavalible(self._hass, self._config.entry_id)
            async_call_later(self._hass, self.client.get_next_reset_delta(), self.reconnect)
        except WebsocketException:
            set_entities_unavalible(self._hass, self._config.entry_id)
            async_call_later(self._hass, DELAY_2, self.reconnect)
```

Reply:
Split into two excepts.